hamt/util.go

```diff
@@ -15,12 +14,21 @@ type hashBits struct {
 	consumed int
 }
 
-func newHashBits(val string) *hashBits {
-	return &hashBits{b: internal.HAMTHashFunction([]byte(val))}
+func newHashBits(val string, hashFunc uint64) *hashBits {
+	return newConsumedHashBits(val, 0, hashFunc)
 }
 
-func newConsumedHashBits(val string, consumed int) *hashBits {
-	hv := &hashBits{b: internal.HAMTHashFunction([]byte(val))}
+func newConsumedHashBits(val string, consumed int, hashFunc uint64) *hashBits {
+	var hashed []byte
+	switch hashFunc {
+	case mh.MURMUR3_128:
+		hashed = murmur3Hash([]byte(val))
+	case mh.IDENTITY:
+		hashed = idHash([]byte(val))
```

Comment:
Putting test code into the main code like this is confusing and makes it hard when people try and audit what's going on here.

For example, this could just take in a function as the hashFunc instead of passing the code around.